src/com/facebook/buck/apple/AppleBundle.java

```diff
@@ -670,6 +672,24 @@ public boolean isLegacyWatchApp() {
     return stepsBuilder.build();
   }
 
+  private void verifyResourceConflicts(AppleBundleResources resources,
+                                       SourcePathResolver resolver) {
+    // Ensure there are no resources that will overwrite each other
+    Set<Path> filenames = new HashSet<>();
+    for (SourcePath path : Iterables.concat(
+        resources.getResourceDirs(),
+        resources.getDirsContainingResourceDirs(),
```

Comment:
This check won't work correctly because it's the contents *inside* that get copied to the root of the resources dir, not the dirs themselves.

The deeper issue here is that those dirs will usually not have the contents at this point but only be generated later (e.g., a `genrule`). So you won't be able to perform the check here.

Given the above, you'd have to move the check to be part of the `CopyStep` itself.

Reply:
I was a bit confused about `DirsContainingResourceDirs`, is this only the output of genrules or also other stuff? How can I set up a BUCK config to check this?

Reply:
If we move this to `CopyStep`, do you want this to be enabled globally or just for apple bundle resources?

Reply:
@rmaz I had to look into it and it seems it's only used for `js_bundle` (see `JsBundleDescription::addAppleBundleResources()`). I couldn't find any integration tests for `js_bundle`, so I think we can ignore `DirsContainingResourceDirs` for now (but please put a `TODO`).